# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/serviceworkers/ServiceWorkerRegistrationProxy.h
# Commit: 1c6ace5aa850
# Full Hash: 1c6ace5aa8502311912112a4a8d80739253579a2
# Author: Kagami Sascha Rosylight <saschanaz@outlook.com>
# Date: 2025-07-16 21:45:35
# Regressor Bug: 1972117
# File Overlap Count: 1
# Description:
#   Bug 1972117 - Part 3: Move NotificationsCallback to NotificationUtils r=asuth
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D255860
# ==============================================================================

diff -r 5db864f2389f -r 1c6ace5aa850 dom/serviceworkers/ServiceWorkerRegistrationProxy.h
--- a/dom/serviceworkers/ServiceWorkerRegistrationProxy.h	Wed Jul 16 15:33:00 2025 +0000
+++ b/dom/serviceworkers/ServiceWorkerRegistrationProxy.h	Wed Jul 16 15:33:00 2025 +0000
@@ -9,6 +9,7 @@
 
 #include "mozilla/dom/ClientInfo.h"
 #include "mozilla/dom/PServiceWorkerRegistrationParent.h"
+#include "mozilla/dom/notification/NotificationUtils.h"
 #include "nsProxyRelease.h"
 #include "ServiceWorkerRegistrationDescriptor.h"
 #include "ServiceWorkerRegistrationListener.h"
@@ -86,7 +87,8 @@
 
   RefPtr<NavigationPreloadStatePromise> GetNavigationPreloadState();
 
-  RefPtr<NotificationsPromise> GetNotifications(const nsAString& aTag);
+  RefPtr<notification::NotificationsPromise> GetNotifications(
+      const nsAString& aTag);
 
   NS_INLINE_DECL_THREADSAFE_REFCOUNTING(ServiceWorkerRegistrationProxy,
                                         override);